packages/eas-cli/src/build/build.ts

```diff
@@ -133,9 +133,11 @@ export async function prepareBuildRequestForPlatformAsync<
 
   let projectArchive: ArchiveSource | undefined;
   if (ctx.localBuildOptions.localBuildMode === LocalBuildMode.LOCAL_BUILD_PLUGIN) {
+    const projectPath = (await makeProjectTarballAsync(ctx.vcsClient)).path;
     projectArchive = {
       type: ArchiveSourceType.PATH,
-      path: (await makeProjectTarballAsync(ctx.vcsClient)).path,
+      path: projectPath,
+      metadataLocation: (await makeProjectMetadataFileAsync(projectPath)).path,
```

Comment:
How about we extract it to `const metadataPath` and use in `projectArchive` like this?

Reply:
Do you mean:
```
const metadataPath = (await makeProjectMetadataFileAsync(projectPath)).path;
// ...
metadataLocation: metadataPath
```
?

Reply:
Yup. I still think something like
```ts
} else if (!ctx.localBuildOptions.localBuildMode) {
  const bucketKey = await uploadProjectAsync(ctx);
  const metadataBucketKey = await uploadMetadataFileAsync({ ctx, projectTarball }); // string | null
  projectArchive = {
    type: 'GCS',
    bucketKey,
    metadataBucketKey,
  };
```

would read better, but it's only my opinion. To me, metadata file feels like something separate from the "project", so it shouldn't live in `uploadProject`. On the other hand it already is part of `projectArchive` type, so one could argue that it _is_ part of the `project`.

Either way, my suggestion would be to make keys of `projectArchive` explicit, not implicit (replace spreading for explicit `const { bucketKey, ... } = await` + `{ ..., bucketKey, ... }`.